src/events/rawGuildMemberUpdate.ts

```diff
@@ -46,22 +49,21 @@ export default class extends Event {
 		// If the user does not have multiple roles from any set cancel
 		if (!hasMultipleRolesInOneSet) return;
 
-		const auditLogs = await api(this.client)
-			.guilds(guild.id)
-			['audit-logs'].get<AuditLogResult>({
-				query: {
-					limit: 10,
-					action_type: 25
-				}
-			});
+		const query: RESTGetAPIAuditLogQuery = {
+			limit: 10,
+			action_type: AuditLogEvent.MEMBER_ROLE_UPDATE
+		};
+		const auditLogs = await api(this.client).guilds(guild.id)['audit-logs'].get<RESTGetAPIAuditLogResult>({
+			query
+		});
 
 		const entry = auditLogs.audit_log_entries.find(
-			(e) => e.user_id !== CLIENT_ID && e.target_id === data.user!.id && e.changes.find((c) => c.key === '$add' && c.new_value.length)
+			(e) => e.user_id !== CLIENT_ID && e.target_id === data.user!.id && e.changes?.find((c) => c.key === '$add')
 		);
 		if (typeof entry === 'undefined') return;
 
-		const change = entry.changes.find((c) => c.key === '$add' && c.new_value.length)!;
-		const updatedRoleID = change.new_value[0].id;
+		const change = entry.changes!.find((c) => c.key === '$add') as APIAuditLogChangeKey$Add;
```

Comment:
Do you NEED this cast? Considering the changes are now..type safe, this cast seems redundant

Also double find

Reply:
Yes, because `Array<T>.find` returns `Array<T>`.

I don't know how to avoid the double find other than moving this to another method and return it as soon as it's found.

Reply:
`find` shouldn't and simply can't return Array<T> but T...
![image](https://user-images.githubusercontent.com/17960496/93373916-b2498b80-f85e-11ea-8782-08a47f6bce23.png)

Reply:
Fixed in 936f92a8.